interfaces/Portalicious/src/app/pages/project-registration-personal-information/components/edit-personal-information/edit-personal-information.component.ts

```diff
@@ -0,0 +1,244 @@
+import { InputTextModule } from 'primeng/inputtext';
+import { SelectModule } from 'primeng/select';
+import { Subscription } from 'rxjs';
+
+import { ConfirmationDialogComponent } from '~/components/confirmation-dialog/confirmation-dialog.component';
+import { FormFieldWrapperComponent } from '~/components/form-field-wrapper/form-field-wrapper.component';
+import { ProjectApiService } from '~/domains/project/project.api.service';
+import { RegistrationApiService } from '~/domains/registration/registration.api.service';
+import {
+  NormalizedRegistrationAttribute,
+  RegistrationAttributeService,
+} from '~/services/registration-attribute.service';
+import { ToastService } from '~/services/toast.service';
+import { generateFieldErrors } from '~/utils/form-validation';
+
+type EditPersonalInformationFormGroup =
+  (typeof EditPersonalInformationComponent)['prototype']['formGroup'];
+
+@Component({
+  selector: 'app-edit-personal-information',
+  imports: [
+    ReactiveFormsModule,
+    FormFieldWrapperComponent,
+    DatePickerModule,
+    ButtonModule,
+    SelectModule,
+    InputTextModule,
+    InputNumberModule,
+    CheckboxModule,
+    ConfirmationDialogComponent,
+    FormsModule,
+  ],
+  templateUrl: './edit-personal-information.component.html',
+  styles: ``,
+  providers: [ToastService],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EditPersonalInformationComponent implements OnInit, OnDestroy {
+  readonly projectId = input.required<string>();
+  readonly registrationId = input.required<string>();
+  readonly attributeList = input.required<NormalizedRegistrationAttribute[]>();
+  readonly cancelEditing = output();
+  readonly registrationUpdated = output();
+
+  readonly projectApiService = inject(ProjectApiService);
+  readonly registrationApiService = inject(RegistrationApiService);
+  readonly registrationAttributeService = inject(RegistrationAttributeService);
+  readonly toastService = inject(ToastService);
+
+  project = injectQuery(this.projectApiService.getProject(this.projectId));
+  registration = injectQuery(
+    this.registrationApiService.getRegistrationById(
+      this.projectId,
+      this.registrationId,
+    ),
+  );
+
+  readonly editPersonalInformationDialog =
+    viewChild.required<ConfirmationDialogComponent>(
+      'editPersonalInformationDialog',
+    );
+  readonly unsavedChangesDialog =
+    viewChild.required<ConfirmationDialogComponent>('unsavedChangesDialog');
+
+  formGroup!: FormGroup<
+    Record<string, FormControl<boolean | number | string | undefined>>
+  >;
+  formGroupChangesSubscription: Subscription;
+  changedRegistrationData = signal<
+    Record<string, boolean | number | string | undefined>
+  >({});
+  updateReason = model<string>();
+
+  ngOnInit() {
+    this.formGroup = this.registrationAttributeService.attributesToFormGroup({
+      attributes: this.attributeList(),
+      projectId: this.projectId(),
+    });
+
```

Comment:
Would it be possible to add something like:

```
  this.router.events
    .pipe(filter((e) => e instanceof NavigationStart))
    .subscribe((event: NavigationStart) => {
      return event.url !== this.router.url;
    });
```

Reply:
Just tried it, but it doesn't work. I implmented a "pendingChangesGuard" instead, based on the description in the relevant task in the PBI. It's in a separate comment.

Reply:
Aha, preventing it "from the other side" makes sense.. and is even cleaner and more reusable 🥳